spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -130,4 +149,43 @@ public static String sourceName(String target, String applicationName, String na
 		return target + PROPERTY_SOURCE_NAME_SEPARATOR + applicationName + PROPERTY_SOURCE_NAME_SEPARATOR + namespace;
 	}
 
+	public static final class Prefix {
```

Comment:
this is a core change when adding support for prefix for labeled sources. First of all it is a nested class on purpose, so that I could access its `private` methods inside the parent class, but not expose them, not even to the package. 
This is needed so that any external interaction is only via its predefined instances.

The need for such a class comes when you think about sources that we find by labels. Let's take a secret for example. Users can specify searching for this secret by some labels, at the same time specifying that they want "useNameAsPrefix". Well, we will only know the name, once we find this source - at this point in the flow, we simply have no idea what the name will be. As such, I call this a "DELAYED" prefix, meaning - I know for sure that I need such a prefix, but I do not know its name _currently_. 

This change, in turn, triggers many tests to be updated, unfortunately.